clickhouse-admin/tests/integration_test.rs

```diff
@@ -197,3 +200,83 @@ async fn test_keeper_conf_parsing() -> anyhow::Result<()> {
     logctx.cleanup_successful();
     Ok(())
 }
+
+#[tokio::test]
+async fn test_keeper_cluster_membership() -> anyhow::Result<()> {
+    let logctx = test_setup_log("test_keeper_cluster_membership");
+    let log = logctx.log.clone();
+
+    let (parent_dir, prefix) = log_prefix_for_test(logctx.test_name());
+    let path = parent_dir.join(format!("{prefix}-oximeter-clickward-test"));
+    std::fs::create_dir(&path)?;
+
+    // We spin up several replicated clusters and must use a
+    // separate set of ports in case the tests run concurrently.
+    let base_ports = BasePorts {
+        keeper: 30500,
+        raft: 30600,
+        clickhouse_tcp: 30700,
+        clickhouse_http: 30800,
+        clickhouse_interserver_http: 30900,
+    };
+
+    let config = DeploymentConfig {
+        path: path.clone(),
+        base_ports,
+        cluster_name: "oximeter_cluster".to_string(),
+    };
+
+    let mut deployment = Deployment::new(config);
+
+    let num_keepers = 3;
+    let num_replicas = 1;
+    deployment
+        .generate_config(num_keepers, num_replicas)
+        .context("failed to generate config")?;
+    deployment.deploy().context("failed to deploy")?;
+
+    wait_for_keepers(
+        &log,
+        &deployment,
+        (1..=num_keepers).map(clickward::KeeperId).collect(),
+    )
+    .await?;
+
+    let clickhouse_cli = ClickhouseCli::new(
+        Utf8PathBuf::from_str("clickhouse").unwrap(),
+        SocketAddrV6::new(Ipv6Addr::LOCALHOST, 30501, 0, 0),
+    )
+    .with_log(log.clone());
+
+    let keeper_cluster_membership =
+        clickhouse_cli.keeper_cluster_membership().await.unwrap();
+
+    let mut raft_config = BTreeSet::new();
+
+    for i in 1..=num_keepers {
+        raft_config.insert(clickhouse_admin_types::KeeperId(i));
+    }
+
+    let expected_keeper_cluster_membership =
```

Comment:
Personally, I wouldn't even bother defining the whole struct just to pull out the 2 fields we care about. I'd just compare what we get back to the values directly.